raiden-dapp/src/service-worker/assistant.ts

```diff
@@ -57,37 +66,41 @@ export default class ServiceWorkerAssistant {
   }
 
   private onMessage(event: MessageEvent): void {
-    if (!(event.data && event.data.messageIdentifier)) return;
+    const message = new ServiceWorkerMessageEvent(event);
 
-    const { messageIdentifier, ...payload } = event.data;
+    // Skip old message formats. A client (us here) can never be connected to
+    // a worker of an older version than the client itself. Thereby the worker
+    // will send the message also in the new and more enriched format again.
+    // This is different for the worker which must expect to receive message
+    // from an client older then himself.
+    if (message.isInOldFormat) return;
 
-    switch (messageIdentifier) {
-      case ServiceWorkerMessageIdentifier.INSTALLED_VERSION:
-        this.setInstalledVersion(payload.version);
-        break;
-
-      case ServiceWorkerMessageIdentifier.INSTALLATION_ERROR:
-        this.reportInstallationError(payload.error);
-        break;
-
-      case ServiceWorkerMessageIdentifier.CACHE_IS_INVALID:
-        this.setUpdateIsMandatory();
-        break;
+    this.messageHandlers[message.type]?.(message.payload);
```

Comment:
You sure this will call the method with `this` set? The methods make use of `this`, and they're set in the mapping; I'm not sure this will preserve `this` in the context, maybe it'd be better to `.call` just to be on the safe side

Reply:
Oh, right, just noticed those are arrow functions; still, it'd be good to ensure which `this` is persisted when the arrow functions are defined in the body of classes

Reply:
They are actually arrow functions in first place so I can say they are of that specific type...

Reply:
So what would you recommend to do? 🤔

Reply:
As you need to specify the function type explicitly in order to ensure it matches a given protocol, the arrow function is fine. Maybe just add the `this` type specifier in the `ServiceWorkerMessageHandler` type, and use `.call` to make explicit your intention/API here, even though they shouldn't matter much with arrow funcitons.

Reply:
Do `.call(this, message.payload)`?

Reply:
Alright, I'll do. Let's make it bulletproof👍🏾

Reply:
I was so rude and amended the last commit and force pushed. I was too lazy to push it as fixup commit and then rebase again etc.
Done. ✔️